sirepo/raydata/scan_monitor.py

```diff
@@ -0,0 +1,522 @@
+                scan
+                for uid, scan in databroker.catalog[catalog_name]
+                .search(databroker.queries.TimeRange(since=m.start()))
+                .items()
+                if uid != m.uid()
+            ]
+            r.sort(key=lambda x: _Metadata(x).start())
+            for s in r:
+                if _Metadata(s).is_scan_plan_executing():
+                    return l
+                _queue_for_analysis(s, catalog_name)
+                l = s
+            return l
+
+        async def _poll_for_new_scans(most_recent):
+            while True:
+                most_recent = _collect_new_scans_and_queue(most_recent)
+                await pkasyncio.sleep(2)
+
+        pkdlog("catalog_name={}", catalog_name)
+        c = None
+        while not c:
+            try:
+                c = databroker.catalog[catalog_name]
+            except KeyError:
+                pkdlog(f"no catalog_name={catalog_name}. Retrying...")
+                await pkasyncio.sleep(15)
+        s = c[-1]
+        if not _Analysis.have_analyzed_scan(_Metadata(s).uid(), catalog_name):
+            _queue_for_analysis(s, catalog_name)
+        await _poll_for_new_scans(s)
+        raise AssertionError("should never get here")
+
+    def _queue_for_analysis(scan, catalog_name):
+        s = PKDict(
+            uid=scan if isinstance(scan, str) else _Metadata(scan).uid(),
+            catalog_name=catalog_name,
+        )
+        pkdlog("scan={}", s)
+        if s not in _SCANS_AWAITING_ANALYSIS:
+            pkio.unchecked_remove(_Analysis.analysis_output_dir(s.uid).join("*"))
+            _SCANS_AWAITING_ANALYSIS.append(s)
+            _Analysis.set_scan_status(s, _AnalysisStatus.PENDING)
+
+    def _scan_info(catalog, scan_uuid, scans_data, status=None):
+        m = _Metadata(catalog[scan_uuid])
+        d = PKDict(
+            uid=scan_uuid,
+            status=status,
+            pdf=True if _analysis_pdf_paths(scan_uuid) else False,
+        )
+        for c in _DEFAULT_COLUMNS:
+            d[c] = getattr(m, c)()
+
+        for c in scans_data.get("selectedColumns", []):
+            d[c] = m.get_scan_field(c)
+        return d
+
+    def _scan_info_result(scans):
+        return PKDict(
+            data=PKDict(
+                scans=sorted(scans, key=lambda e: e.start),
+                cols=[k for k in scans[0].keys() if k not in _NON_DISPLAY_SCAN_FIELDS]
+                if scans
+                else [],
+            )
+        )
+
+    global cfg
+    cfg = pkconfig.init(
+        catalog_names=(frozenset(), set, "list of catalog names to monitor"),
+        concurrent_analyses=(
+            2,
+            int,
+            "max number of analyses that can run concurrently",
+        ),
+        db_dir=pkconfig.RequiredUnlessDev(
+            os.path.join(sirepo.srdb.root(), "raydata"),
+            pkio.py_path,
+            "root directory for db",
+        ),
+        notebook_dir=pkconfig.RequiredUnlessDev(
```

Comment:
This can be just `notebook_dir=pkconfig.Required(pkio.py_path, "directory for analysis notebooks")`. We can expect all dev users to run using the script and in higher environments it will be configured.

Reply:
b2193636ae2f3070921e6c0eabdce8ec9b2fdc8b